test/fs/test_stat_unnamed_file_descriptor.c

```diff
@@ -0,0 +1,40 @@
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <assert.h>
+#include "stdio.h"
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#endif
+
+void makedir(const char *dir) {
+  int rtn = mkdir(dir, 0777);
+  assert(rtn == 0);
+}
+
+void changedir(const char *dir) {
+  int rtn = chdir(dir);
+  assert(rtn == 0);
+}
+
+
+int main() {
+    makedir("working");
+#if defined(__EMSCRIPTEN__) && defined(NODEFS)
+    EM_ASM(FS.mount(NODEFS, { root: '.' }, 'working'));
+#endif
+    changedir("working");
```

Comment:
Can you put the makedir and changedir calls (and the function defs inside the `defined(__EMSCRIPTEN__) && defined(NODEFS)` block?

Also maybe move this code into a separate `setup()` function?